### PR TITLE
Plans CRUD - R

### PR DESCRIPTION
**Pull request by ianlessa**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| What?         | Plans CRUD - R
| Why?          | Enable plan, single and normal product filtering.
| How?          | Added a custon filter and override Opencart default methods/javascripts.

### CODE REVIEW

File: system/library/mundipagg/src/Controller/Events.php [hunk 1]
```diff
@@ -240,6 +234,68 @@ protected function handleProductDelete()
...
     }
 
+    protected function handleProductIndexError($errorData)
+    {
+        $opencartReflection = new \ReflectionClass($this->openCart);
+        $errorProperty = $opencartReflection->getProperty('error');
+        $errorProperty->setAccessible(true);
+        $currentErrors = $errorProperty->getValue($this->openCart);
+
+        $currentErrors = array_merge($currentErrors,$errorData);
```
Comment (GabrielDeveloper) [inline system/library/mundipagg/src/Controller/Events.php]: space

File: system/library/mundipagg/src/Controller/Events.php [hunk 2]
```diff
@@ -240,6 +234,68 @@ protected function handleProductDelete()
...
+        $registryProperty->setAccessible(false);
+
+        $file = DIR_APPLICATION . 'controller/catalog/product.php';
+        require_once($file);
+        $productController = new \ControllerCatalogProduct($registry);
+
+        $productControllerReflection = new \ReflectionClass($productController);
+        $errorProperty = $productControllerReflection->getProperty('error');
+        $errorProperty->setAccessible(true);
+        $errorProperty->setValue($productController,$currentErrors);
```
Comment (GabrielDeveloper) [inline system/library/mundipagg/src/Controller/Events.php]: space

File: system/library/mundipagg/src/Controller/Events.php [hunk 3]
```diff
@@ -240,6 +234,68 @@ protected function handleProductDelete()
...
+        $productControllerReflection = new \ReflectionClass($productController);
+        $productController->index();
+
+        $this->openCart->load->model('extension/payment/mundipagg_product');
+
+        $registryProperty = $productControllerReflection->getProperty('registry');
+        $registryProperty->setAccessible(true);
+        $registry = $registryProperty->getValue($productController);
+        $registry->set('model_catalog_product',$this->openCart->model_extension_payment_mundipagg_product);
+        $registryProperty->setValue($productController,$registry);
```
Comment (GabrielDeveloper) [inline system/library/mundipagg/src/Controller/Events.php]: space

File: admin/model/extension/payment/mundipagg_product.php [hunk 4]
```diff
@@ -0,0 +1,96 @@
+<?php
+
+class ModelExtensionPaymentMundipaggProduct extends ModelCatalogProduct
+{
+    public function getProducts($data = array())
+    {
+        $filter = isset($this->request->get['filter_mp_type']) ?
```
Comment (michelpl) [inline admin/model/extension/payment/mundipagg_product.php]: Se tem que quebrar linha, não pode ser ternário

**Pull request merged**